torch/distributed/elastic/multiprocessing/api.py

```diff
@@ -470,11 +471,12 @@ def __init__(
 
     def start(self) -> None:
         """Start processes using parameters defined in the constructor."""
-        signal.signal(signal.SIGTERM, _terminate_process_handler)
-        signal.signal(signal.SIGINT, _terminate_process_handler)
-        if not IS_WINDOWS:
-            signal.signal(signal.SIGHUP, _terminate_process_handler)
-            signal.signal(signal.SIGQUIT, _terminate_process_handler)
+        if threading.current_thread() is threading.main_thread():
```

Comment:
We should have a proper error message/warning when something like this occurs since it can cause unexpected behavior.